### PR TITLE
updated R1 variable ToroidalFieldCoilPrincetonD

### PR DESCRIPTION
**Pull request by billingsley-john**
## Proposed changes

PR updating the R1 variable of the TFCoilPrincetonD to correspond to the radial distance from x=0 to the first layer of the inner leg of the tf coil.

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by billingsley-john** [comment]
This has been achieved by simply adding + self.thickness to the arguments of self._compute_inner_points(); however, this can cause confusion as R1 within that method is no longer equal to self.R1 - perhaps this method should be updated to use a different variable name.

**Comment by shimwell** [comment]
I wonder if it is worth making the ITERTokamak and  EuDemoFrom2015PaperDiagram to check their TF coils are and casing are still working

**Comment by billingsley-john** [comment]
> I wonder if it is worth making the ITERTokamak and EuDemoFrom2015PaperDiagram to check their TF coils are and casing are still working

Indeed, ITERTokamak will need updating

**Review by shimwell — APPROVED** [review]

**Comment by billingsley-john** [comment]
As vertical_section_offset can now be smaller than outer_offset, the method of construction where the outer coil points are translated by outer_offset and unioned with the inner leg means excess at the back of the casing can exist (see image). Previously, when the inner radius of the inner leg was determined by outer_offset + vertical_section_offset, this would not happen as the translated outer points would always be within the inner leg.

Screenshot from 2021-03-26 15-59-21

This can be solved by cutting away the excess coil.

**Comment by billingsley-john** [comment]
New functionality

`
casing = paramak.TFCoilCasing(magnet=magnet, inner_offset=10, outer_offset=30, vertical_section_offset=20, distance=40)
`

Screenshot from 2021-03-26 16-15-29

**Comment by billingsley-john** [comment]
Not the most elegant solution so any suggestions welcome!

**Comment by billingsley-john** [comment]
TODO:
Add tests

**Comment by billingsley-john** [comment]
TFCoilPrincetonD

**Comment by shimwell** [comment]
Looks like there is just one tiny code inspector comment

File: paramak/parametric_components/tf_coil_casing.py [hunk 1]
```diff
@@ -100,5 +141,13 @@ def create_solid(self):
             self.azimuth_placement_angle
         solid = union_solid(solid, self.leg_shape)
         solid = cut_solid(solid, self.magnet)
+
+        inner_bore_cutting_shape = ExtrudeStraightShape(
```
Comment (RemDelaporteMathurin) [inline paramak/parametric_components/tf_coil_casing.py:148]: @billingsley-john perhaps some comments could help understand what you did here (nice find btw)

Comment (RemDelaporteMathurin) [reply paramak/parametric_components/tf_coil_casing.py:148]: maybe even refer to the issue

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]

**Comment by RemDelaporteMathurin** [comment]
Nice find @billingsley-john. If I understand well, you have an additionnal cut operation to get rid of any potential excess right ?

File: paramak/parametric_components/tf_coil_casing.py (hunk 1 above)
Comment (billingsley-john) [reply paramak/parametric_components/tf_coil_casing.py:148]: Sounds great, I shall add some!

**Comment by billingsley-john** [comment]
> Nice find @billingsley-john. If I understand well, you have an additionnal cut operation to get rid of any potential excess right ?

Yes, there is sometimes excess magnet depending on the outer_offset and vertical_section_offset values. This cuts away this excess. Before merging, we should test the component is still created successfully when there is no excess. I'm not entirely sure of the behaviour of CadQuery when a cut is attempted but the solids are not overlapping. Perhaps this has to be guarded against (i.e. only perform a cut if there is excess)

**Review by RemDelaporteMathurin — APPROVED** [review]

**Comment by billingsley-john** [comment]
Now uses coil workplane to create casing in correct workplane

**Pull request merged**